MH-13205 Document, fix and improve thumbnail support

staubesv opened the pull request:
Oh man... Actually, I wanted to implement [MH-13201
Convert uploaded images to appropriate size and format](https://opencast.jira.com/browse/MH-13201), but stumbled over [MH-13204 The uploaded thumbnail needs to be tagged with 'archive'](https://opencast.jira.com/browse/MH-13204).

Finally ended up with this:

This task is about addressing a number of related issues considering Opencast 6.x new thumbnail feature:

### Archive tag for the uploaded thumbnail (bugfix)
When a user uploads a thumbnail image, it needs to be made persistent. The current implementation missed to add the tag 'archive' to the attachment added to the media package. Without that tag, the Opencast default workflows will not include the uploaded thumbnail in snapshots they create.
As it would be somewhat nasty to hard-code this, I've added the configuration property thumbnail.uploaded.tags.

**Untangle the flavor of the source track from the flavor of the uploaded thumbnail (enhancement)**
The flavor of the attachment created for the uploaded thumbnail was determined by thumbnail.source.flavor.type and thumbnail.source.flavor.subtype. This mixes up the identification of the source track with the flavor of the uploaded thumbnail which is confusing and limiting.
I've introduced thumbnail.uploaded.flavor to make break up the undesirable entanglement. The configuration property thumbnail.source.flavor.type could then be removed as not used otherwise.

**Document the thumbnail configuration (documentation)**
As there are quite a number of configuration properties related to configure the thumbnail feature, I've felt like there should be some documentation about this.
I've refactored the existing documentation related to the Admin UI in the Admin Guide and added a page describing how the configuration properties for the thumbnail feature work.

**Don't hard-code encoding profiles (enhancement)**
During all the work, I've found that the encoding profile used to downscale the thumbnail preview image is hardcoded which is undesirable and weird to document. I therefore introduce the new configuration property thumbnail.preview.profile.downscale.

**Renamed some configuration properties (maintenance)**
I've renamed thumbnail.default.track.primary and thumbnail.default.track.secondary to thumbnail.source.flavor.type.primary and thumbnail.source.flavor.type.secondary for two reasons: Those configuration keys are used together with thumbnail.source.flavor.subtype to create the source flavors. Also, those configuration properties are currently not only used for the default thumbnail (even though this is not necessarily a great things...).

Overview of changes to the structure of the documentation:

![screen shot 2018-11-01 at 17 30 19](https://user-images.githubusercontent.com/1590263/47865207-e26b1b00-ddfb-11e8-9bb0-3e5d3b9a30ff.png)

As we are getting more documentation about the configuration of the Admin UI, I have restructured the documentation:

- Introduction of the subfolder configuration/admin-ui to group pages that related to admin ui configuration
- Move "Manuel Asset Upload" to that subfolder (this is admin ui configuration)
- Splitted up the existing configuration/ui.md (Event Filters, Languages, Metadata parts moved to metadata configuration page)
- Finally... added page admin-ui/thumbnails.md

staubesv committed:
MH-13205 Document, fix and improve thumbnail support

- Add tag 'archive' (configurable) to uploaded thumbnail so it is included in future snapshots
- Untangle the flavor of the source track from the flavor of the uploaded thumbnail
- Document the thumbnail configuration
- Don't hard-code encoding profile
- Rename some configuration properties

staubesv committed:
Remove trailing whitespace

staubesv committed:
Remove trailing whitespace

staubesv commented on docs/guides/admin/docs/configuration/admin-ui/languages.md:
```diff
@@ -0,0 +1,5 @@
+Language Configuration
```
Moved from configuration/ui.md

staubesv commented on docs/guides/admin/docs/configuration/admin-ui/thumbnails.md:
```diff
@@ -0,0 +1,156 @@
+Overview
```
This page is new

gregorydlogan replied:
This file doesn't mention which configuration file(s) we're documenting.

staubesv replied:
Added.

staubesv commented on docs/guides/admin/docs/configuration/metadata.md:
```diff
@@ -142,3 +142,40 @@ tagging operation for dublincore metadata the following way
       </configurations>
     </operation>
+
+## Configuring the events publisher metadata field
```
Moved from configuration/ui.md (the metadata configuration has more impact than just admin ui)

staubesv commented on docs/guides/admin/docs/configuration/admin-ui/event-filters.md:
```diff
@@ -1,8 +1,5 @@
-
-Configuring the events filters
---------------------------------
+Events Filters Configuration
```
Reduced configuration/ui.md to "Event Filters Configuration" page configuration/admin-ui/event-filters.md. All other content has been moved to other places.

gregorydlogan replied:
Loading this page generates a 404, which is probably caused by a bad filename:
```
...
[I 181105 10:52:06 handlers:79] Ignore: /home/greg/opencast/upstream/docs/guides/admin/docs/configuration/admin-ui/images/assetUploadExistingEvent.png
...
[I 181105 10:52:23 handlers:132] Browser Connected: http://localhost:8000/configuration/admin-ui/event-filters/
[W 181105 10:52:37 web:2064] 404 GET /configuration/admin-ui/assetUploadSource.png (127.0.0.1) 6.85ms
...
```

staubesv replied:
Fixed.

staubesv commented on docs/guides/admin/docs/configuration/admin-ui/event-filters.md:
```diff
@@ -70,49 +67,3 @@ currently not possible.
 user's screen.
-
-
-Available Language Configuration
```
Moved to new page configuration/admin-ui/languages.md

staubesv commented on docs/guides/admin/docs/configuration/admin-ui/event-filters.md:
```diff
@@ -70,49 +67,3 @@ currently not possible.
-your users, add the relevant locale code to `etc/org.opencastproject.adminui.endpoint.LanguageServiceEndpoint.cfg`.
-
-
-Configuring the events publisher metadata field
```
Moved to existing page configuration/metadata.md

staubesv commented on docs/guides/admin/docs/configuration/workflow.md:
```diff
@@ -335,62 +335,8 @@ Thumbnail Type | Description
 1              | The thumbnail has been uploaded and is stored in the asset manager as media package attachment
 2              | The thumbnail shall be extracted at a given time position from a given track
 
-At any given point of time, there is exactly one thumbnail available for each published event.
```
Avoid too much duplication -> refers to configuration/admin-ui/thumbnails.md

gregorydlogan commented on docs/guides/admin/docs/configuration/admin-ui/thumbnails.md:
```diff
@@ -0,0 +1,156 @@
+In this example, the default thumbnail would be extracted from the track with the flavor `presenter/source` or, if
+no such track is available, a track with the flavor `presentation/source`.
+
+The relative position within the cutted video where the default thumbnail is extracted can be configured:
```
`Within the cutted video` -> `within the final, edited video`

staubesv replied:
Changed to "edited video". Not that this is not necessarily the "final" video. For example, the final video might include an intro video.

gregorydlogan commented on docs/guides/admin/docs/configuration/admin-ui/thumbnails.md:
```diff
@@ -0,0 +1,156 @@
+
+Opencast will set the following processing settings for the event being edited: `thumbnailType` is set to `0`
+to indicate that the default thumbnail is used. `thumbnailPosition` is set to the absolute position of the
+uncutted video where the default thumbnail should be extracted.
```
edited video

staubesv replied:
Just omitted "uncutted". So "edited video" vs "video" now.

gregorydlogan commented on docs/guides/admin/docs/configuration/admin-ui/thumbnails.md:
```diff
@@ -0,0 +1,156 @@
+------------------
+
+In case the user is not happy with the automatically generated default thumbnail, the user can extract a thumbnail
+at an arbitrary position withhin the uncutted version of the video.
```
`withhin` -> `within`, `uncutted` -> `unedited`

staubesv replied:
Fixed "within", removed "uncutted"

gregorydlogan commented on docs/guides/admin/docs/configuration/admin-ui/thumbnails.md:
```diff
@@ -0,0 +1,156 @@
+    # Default: archive
+    #thumbnail.uploaded.tags=archive
+
+**IMPORTANT:** Please ensure that all workflows in your setup will always included this attachment in their snapshots.
```
An example of how to do this should be added IMO.  Which flavour(s) and/or tags need to be added, and to which operation(s), assuming we're using the default workflow.

gregorydlogan commented on docs/guides/admin/docs/configuration/admin-ui/thumbnails.md:
```diff
@@ -0,0 +1,156 @@
+    # Default: 1.0
+    #thumbnail.default.position=1.0
+
+Opencast will set the following processing settings for the event being edited: `thumbnailType` is set to `0`
```
This reads like it should be a point form list, but does not render as such.  IMO a point form list makes sense here.  This comment applies to all four thumbnail types.

staubesv replied:
Agree, changed to point form list for all thumbnail types.

gregorydlogan commented on docs/guides/admin/docs/configuration/admin-ui/thumbnails.md:
```diff
@@ -0,0 +1,156 @@
+Opencast supports automatic distribution of thumbnail images for publication channels that support incremental
+publication.
+
+Currenlty, this is supported for External API publication channels (created by WOH publish-configure) and OAI-PMH
```
Currently

staubesv replied:
Fixed.

staubesv committed:
MH-13205 Code Review

- Fixed path to image assetUploadSource.png
- Mention configuration file where configuration options for thumbnails can be found in thumbnail documentation
- "cutting video" -> "edited video", "uncutted video" -> "video"
- Use list for enumerating variables that will be set when editing an event
- Be more precise about the information that uploaded thumbnails need to be included in snapshots

staubesv committed:
MH-13205 Code Review (Fixed Typo)

staubesv committed:
MH-13205 Code Review (Fix indentation)

staubesv commented:
That is a fun:

![screen shot 2018-11-06 at 14 02 00](https://user-images.githubusercontent.com/1590263/48066032-b8837100-e1cc-11e8-8882-a2f2031bc2b8.png)

So one warning tells me to not indent the lists and the other says that four spaces are expected ;-)

Went for four spaces, let's see what Travis thinks about that.
